js/TodoUser.js

```diff
@@ -1,46 +1,47 @@
-import { ADDRESS } from './constants.js'
+import {API} from './API.js';
 
-export default function TodoUser ($userTitle, $userList, userName, setActiveUser) {
-  this.$userTitle = $userTitle
-  this.$userList = $userList
-  this.userName = userName
-  this.userList = []
-
-  this.getUsers = () => {
-    fetch(`${ADDRESS.BASE_URL}/api/u`, {
-      method: 'GET'
-    }).then((response) => response.json())
-      .then((data) => {
-        this.userList = data
-        this.render()
-        this.bindEvents()
-      })
-  }
-
-  this.setState = (activeUserName) => {
-    this.userName = activeUserName
-    this.getUsers()
-    setActiveUser(this.userName)
-  }
+export default function TodoUser($userTitle, $userCreateButton, $userList, userId, setActiveUser) {
+  this.$userTitle = $userTitle;
+  this.$userList = $userList;
+  this.$userCreateButton = $userCreateButton;
+  this.userName = '';
+  this.userId = userId;
+  this.userList = [];
+  
+  const onUserCreateHandler = async() => {
+    const userName = prompt('추가하고 싶은 이름을 입력해주세요.');
+    await API.AddUser(userName);
+    this.setState(this.userId);
+  };
+  
+  this.setState = async (activeUserId) => {
+    this.userId = activeUserId;
+    this.userList = await API.GetUsers();
+    this.render();
+    this.bindEvents();
+  };
 
   this.bindEvents = () => {
     document.querySelectorAll('.ripple').forEach(($item) => {
-      $item.addEventListener('click', (e) => {
-        this.userName = e.target.innerText
-        this.setState(this.userName)
-      })
-    })
-  }
+      $item.addEventListener('click', ({target}) => {
+        if (target.classList.contains('user-create-button')) {
+          onUserCreateHandler();
+        } else {
+          this.userName = target.innerText;
+          this.userId = target.id;
```

Comment:
```suggestion
      $item.addEventListener('click', ({target: {classList, innerText }) => {
        if (classList.contains('user-create-button')) {
          onUserCreateHandler();
        } else {
          this.userName = innerText;
          this.userId = target.id;
```
비구조화 할당을 한번 더  써서 객체의 접근자를 줄여서 쓰는 방법도 있습니다